rdt/transformers/base.py

```diff
@@ -9,20 +9,122 @@ class BaseTransformer:
     and ``fit_transform`` method is already implemented.
     """
 
-    def fit(self, data):
+    INPUT_TYPE = None
+    OUTPUT_TYPES = None
+    DETERMINISTIC_TRANSFORM = None
+    DETERMINISTIC_REVERSE = None
+    COMPOSITION_IS_IDENTITY = None
+    NEXT_TRANSFORMERS = None
+
+    @classmethod
+    def get_input_type(cls):
+        """Return the input type supported by the transformer.
+
+        Returns:
+            string:
+                Accepted input type of the transformer.
+        """
+        return cls.INPUT_TYPE
+
+    def get_output_types(self):
+        """Return the output types supported by the transformer.
+
+        Returns:
+            dict:
+                Mapping from the transformed column names to supported data types.
+        """
+        if self.OUTPUT_TYPES:
+            return {f'{self._column_prefix}.{k}': v for k, v in self.OUTPUT_TYPES.items()}
+        
+        return None
+
+    def is_transform_deterministic(self):
+        """Return whether the transform is deterministic.
+
+        Returns:
+            bool:
+                Whether or not the transform is deterministic.
+        """
+        return self.DETERMINISTIC_TRANSFORM
+
+    def is_reverse_deterministic(self):
+        """Return whether the reverse transform is deterministic.
+
+        Returns:
+            bool:
+                Whether or not the reverse transform is deterministic.
+        """
+        return self.DETERMINISTIC_REVERSE
+
+    def is_composition_identity(self):
+        """Return whether composition of transform and reverse transform produces the input data.
+
+        Returns:
+            bool:
+                Whether or not transforming and then reverse transforming returns the input data.
+        """
+        return self.COMPOSITION_IS_IDENTITY
+
+    def get_next_transformers(self):
+        """Return the suggested next transformer to be used for each column.
+
+        Returns:
+            dict:
+                Mapping from transformed column names to the transformers to apply to each column.
+        """
+        if self.NEXT_TRANSFORMERS:
+            return {f'{self._column_prefix}.{k}': v for k, v in self.NEXT_TRANSFORMERS.items()}
+
+        return None
+
+    def fit(self, data, columns):
         """Fit the transformer to the data.
 
         Args:
             data (pandas.Series or numpy.array):
                 Data to transform.
+            columns (list):
+                List of column names from the data.
+        """
+        self._columns = columns
+        self._column_prefix = '#'.join(columns)
+        while self._column_prefix in data: # make sure the _column_prefix doesn't exist in the data
+            self._column_prefix.append('#')
+
+        columns_data = data[columns]
+        self._fit(columns_data)
+
+    def _fit(self, columns_data):
+        """Fit the transformer to the data.
+
+        Args:
+            columns_data (pandas.Series or numpy.array):
```

Comment:
Change this if `fit` only takes `dataframes`.

Reply:
Implemented it. My only concern is that with each new features the code is becoming more complex, so I'll have to do a good job of documenting.